### PR TITLE
[ADD Project Task Date Shift Module

### PR DESCRIPTION
**Pull request by patrickrwilson**
Adds the Project Task Date Shift Module. This module provide the optional ability to shift the dates on tasks when the project start date changes.

Original Discussion #500

### CODE REVIEW

File: project_task_date_shift/README.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+This module is part of the OCA/project on GitHub.
+
+You are welcome to contribute. To learn how please visit https://odoo-community.org/page/Contribute.
```
Comment (emagdalenaC2i) [inline project_task_date_shift/README.rst:82]: You can update README follw the folder readme structure, see https://github.com/OCA/maintainer-tools/tree/master/template/module

File: project_task_date_shift/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+    'name': "Project Task Date Shift",
+
+    'summary': """
+        Project Task Date Shift""",
```
Comment (emagdalenaC2i) [inline project_task_date_shift/__manifest__.py]: Delete extra "
I think it should be 'summary': "Project Task Date Shift",

File: project_task_date_shift/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    'author': "Patrick Wilson, Odoo Community Association (OCA)",
+    'website': "https://github.com/OCA/project",
+
```
Comment (emagdalenaC2i) [inline project_task_date_shift/__manifest__.py]: delete blank lines

File: project_task_date_shift/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    'category': 'Project Management',
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+
+    # any module necessary for this one to work correctly
```
Comment (emagdalenaC2i) [inline project_task_date_shift/__manifest__.py]: delete comments

File: project_task_date_shift/models/project.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
+from dateutil.relativedelta import relativedelta
+from odoo import models, fields, api
+
+
+class Project(models.Model):
+    _inherit = 'project.project'
+
+    old_start_date = fields.Date(string='Old Start Date',
+                                 help="Used by the Shift Dates function. \
```
Comment (emagdalenaC2i) [inline project_task_date_shift/models/project.py]: I think this character is wrong \^M

File: project_task_date_shift/views/project_task.xml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      <data>
+        <xpath expr="//field[@name='date_deadline']" position = "before">
+          <field name="date_start"/>
+          <field name="date_end"/>
+        </xpath>
+      </data>
+    </field>
+  </record>
+
+  <!-- Gantt View -->
```
Comment (emagdalenaC2i) [inline project_task_date_shift/views/project_task.xml]: Gantt View is a Timeline view?

File: project_task_date_shift/models/project.py (hunk 5 above)
Comment (patrickrwilson) [reply project_task_date_shift/models/project.py]: I don't know of another way to break the long help description up to be pep8 compliant and '\' seemed to work for all the other modules I've built. Any other suggestion?

File: project_task_date_shift/views/project_task.xml (hunk 6 above)
Comment (patrickrwilson) [reply project_task_date_shift/views/project_task.xml]: I changed it to  to be more clear.

File: project_task_date_shift/models/project.py (hunk 5 above)
Comment (emagdalenaC2i) [reply project_task_date_shift/models/project.py]: I think you don't need any extra character like in this example:
        help="For records searched by name (taxes, fiscal "
             "positions), the template name will be matched against the "
             "record name on this language.",

Comment (patrickrwilson) [reply project_task_date_shift/models/project.py]: Thanks, i've updated the code and it seems to be working correctly.

**Comment by rafaelbn** [comment]
Please attend to an existing module https://github.com/OCA/project/tree/10.0/project_recalculate

cc @pedrobaeza

**Comment by patrickrwilson** [comment]
@rafaelbn I'm not confident enough that my programming skills are advanced enough to convert that module to v12. If it's preferred by the community to convert that module since it's more advanced with looking at working hours, then feel free to cancel this pr.

**Comment by rafaelbn** [comment]
@pedrobaeza when will we migrate https://github.com/OCA/project/tree/10.0/project_recalculate to v12?

**Comment by pedrobaeza** [comment]
Not yet, but the module should be very similar

**Comment by max3903** [comment]
Shall we close this PR?

**Comment by patrickrwilson** [comment]
I think so, seems it would be best to migrate the older module.

**Pull request closed without merging**